committeeoversightapp/views.py

```diff
@@ -36,78 +45,60 @@ def post(self, request, **kwargs):
-                save_document(url, note, event)
-
-            # find and create witnesses
-            for witness in witness_formset.cleaned_data:
-                name = witness.get('name', None)
-                if name:
-                    # add witness as EventParticipant
-                    entity_type = "person"
-                    note = "witness"
-                    new_witness = EventParticipant(
-                                        name=name,
-                                        event=event,
-                                        entity_type=entity_type,
-                                        note=note
-                    )
-                    new_witness.save()
-
-                    #save witness statement urls TK
-                    witness_url = witness.get('url', None)
-                    witness_document = save_document(witness_url, "witness statement", event)
-
-                    #save witness organizations and link to statement urls
-                    organization = witness.get('organization', None)
-                    retired = witness.get('retired', False)
-                    new_witness_details = WitnessDetails(
-                                        witness=new_witness,
-                                        document=witness_document,
-                                        organization=organization,
-                                        retired=retired
-                    )
-                    new_witness_details.save()
+            save_committees(event, committee_form.cleaned_data['name'])
+            save_category(event, category_form.cleaned_data['category'])
+            save_documents(event, transcript_form.cleaned_data)
+            save_witnesses(event, witness_formset.cleaned_data)
 
             # save event source
             source = EventSource(event=event, note="web form")
             source.save()
 
         return redirect('list-event')
 
-class EventList(LoginRequiredMixin, ListView):
-    model = Event
+class EventList(LoginRequiredMixin, TemplateView):
     template_name = 'list.html'
-    context_object_name = 'hearings'
-    queryset = Event.objects.all().order_by('-updated_at')[:500]
+
+class EventListJson(BaseDatatableView):
+    """ Uses django-datatables-view for server-side DataTable processing."""
+    model = Event
+
+    # define the columns that will be returned
+    columns = ['updated_at', 'name', 'start_date', 'id', 'id']
+
+    # define column names that will be used in sorting
+    # order is important and should be same as order of columns
+    # displayed by datatables. For non sortable columns use empty
+    # value like ''
+    order_columns = ['updated_at', 'name', 'start_date', '', '']
+
+    # max limit of records returned
+    max_display_length = 500
+
+    def prepare_results(self, qs):
+        json_data = []
+        edit_string = "<a href=\"/edit/{}\"><i class=\"fas fa fa-pencil-alt\" id=\"edit-icon\"></i></a>"
```

Comment:
I am not terribly offended by this, actually. Did you search for an alternative? I did a little googling ("pass html string from view to template django"), but did not find anything useful.